PythonAPI/carla/source/carla/libcarla.pyi

```diff
@@ -0,0 +1,5403 @@
+
+    # region Instance Variables
+    @property
+    def speed_kp() -> float:
+        """Proportional term of the speed PID controller."""
+        ...
+
+    @property
+    def speed_ki() -> float:
+        """Integral term of the speed PID controller."""
+        ...
+
+    @property
+    def speed_kd() -> float:
+        """Derivative term of the speed PID controller."""
+        ...
+
+    @property
+    def accel_kp() -> float:
+        """Proportional term of the acceleration PID controller."""
+        ...
+
+    @property
+    def accel_ki() -> float:
+        """Integral term of the acceleration PID controller."""
+        ...
+
+    @property
+    def accel_kd() -> float:
+        """Derivative term of the acceleration PID controller."""
+        ...
+    # endregion
+
+    # region Methods
+    def __init__(self,
+                 speed_kp: float = 0.15,
+                 speed_ki: float = 0.0,
+                 speed_kd: float = 0.25,
+                 accel_kp: float = 0.01,
+                 accel_ki: float = 0.0,
+                 accel_kd: float = 0.01) -> AckermannControllerSettings:
+        """Manages the settings of the Ackermann PID controller.
+        """
+    # endregion
+
+    # region Dunder Methods
+    def __eq__(self, __value: AckermannControllerSettings) -> bool: ...
+
+    def __ne__(self, __value: AckermannControllerSettings) -> bool: ...
+
+    def __str__(self) -> str: ...
+    # endregion
+
+
+class Actor():
+    """CARLA defines actors as anything that plays a role in the simulation or can be moved around.
+
+    That includes: pedestrians, vehicles, sensors and traffic signs (considering traffic lights as part of these). Actors are spawned in the simulation by `carla.World` and they need for a `carla.ActorBlueprint` to be created. These blueprints belong into a library provided by CARLA, find more about them here.
+    https://carla.readthedocs.io/en/latest/bp_library/
+    """
+
+    # region Instnce Variables
+    @property
+    def attributes() -> dict:
+        """A dictionary containing the attributes of the blueprint this actor was based on.
+        """
+        ...
+
+    @property
+    def id() -> int:
+        """Identifier for this actor. Unique during a given episode.
+        """
+        ...
+
+    @property
+    def type_id() -> str:
+        """The identifier of the blueprint this actor was based on,
+
+        e.g., `vehicle.ford.mustang.`"""
+        ...
+
+    @property
+    def is_alive() -> bool:
+        """Returns whether this object was destroyed using this actor handle."""
+        ...
+
+    @property
+    def is_active() -> bool:
+        """Returns whether this actor is active (`True`) or not (`False`)."""
+        ...
+
+    @property
+    def is_dormant() -> bool:
+        """Returns whether this actor is dormant (`True`) or not (`False`) - the opposite of `is_active`."""
+        ...
+
+    @property
+    def parent() -> ["Actor"]:

```

Comment:
`[Type]` is not proper. Should be changed to `Optional["Actor"] `or `Union[None, "Actor"]`